app/js/app/directives/graph_sketcher/GraphSketcher.js

```diff
@@ -1,1087 +1,317 @@
+                                found = true;
+                                p.cursor(p.MOVE);
+                            } else if (detect(c.minX, c.minY) || detect(c.maxX, c.minY) || detect(c.minX, c.maxY) || detect(c.maxX, c.maxY)) {
+                                p.push();
+                                p.fill(graphViewBuilder.graphView.KNOT_DETECT_COLOR);
+                                if (detect(c.minX, c.minY)) {
+                                    p.rect(c.minX - 4, c.minY - 4, 8, 8);
+                                } else if (detect(c.maxX, c.minY)) {
+                                    p.rect(c.maxX - 4, c.minY - 4, 8, 8);
+                                } else if (detect(c.minX, c.maxY)) {
+                                    p.rect(c.minX - 4, c.maxY - 4, 8, 8);
+                                } else {
+                                    p.rect(c.maxX - 4, c.maxY - 4, 8, 8);
                                 }
 
-                                var interX = curve.interX;
-                                freeAllSymbols(interX);
-
-                                var interY = curve.interY;
-                                freeAllSymbols(interY);
+                                p.pop;
 
-                                var maxima = curve.maxima;
-                                freeAllSymbols(maxima);
+                                found = true;
+                                p.cursor(p.MOVE);
+                            } else if (detect((c.minX + c.maxX) / 2, c.minY - 3) || detect((c.minX + c.maxX) / 2, c.maxY + 3)
+                                || detect(c.minX - 3, (c.minY + c.maxY) / 2) || detect(c.maxX + 3, (c.minY + c.maxY) / 2)) {
 
-                                var minima = curve.minima;
-                                freeAllSymbols(minima);
+                                p.push();
+                                p.fill(graphViewBuilder.graphView.KNOT_DETECT_COLOR);
+                                if (detect((c.minX + c.maxX) / 2, c.minY - 3)) {
+                                    p.triangle((c.minX + c.maxX) / 2 - 5, c.minY - 2, (c.minX + c.maxX) / 2 + 5, c.minY - 2, (c.minX + c.maxX) / 2, c.minY - 7);
+                                } else if (detect((c.minX + c.maxX) / 2, c.maxY + 3)) {
+                                    p.triangle((c.minX + c.maxX) / 2 - 5, c.maxY + 2, (c.minX + c.maxX) / 2 + 5, c.maxY + 2, (c.minX + c.maxX) / 2, c.maxY + 7);
+                                } else if (detect(c.minX - 3, (c.minY + c.maxY) / 2)) {
+                                    p.triangle(c.minX - 2, (c.minY + c.maxY) / 2 - 5, c.minX - 2, (c.minY + c.maxY) / 2 + 5, c.minX - 7, (c.minY + c.maxY) / 2);
+                                } else {
+                                    p.triangle(c.maxX + 2, (c.minY + c.maxY) / 2 - 5, c.maxX + 2, (c.minY + c.maxY) / 2 + 5, c.maxX + 7, (c.minY + c.maxY) / 2);
```

Comment:
triangle(...) might be a good candidate for moving into graph view in the future.